### PR TITLE
Feature/tt 33 gas pump

### PR DESCRIPTION
**Pull request by opefago**
Added abi for V2 Custodial Wallet Factory V2

### CODE REVIEW

File: packages/shared/blockchain/evm-based/src/lib/services/evm-based.custodial.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ enum ContractType {
   SEMI_FUNGIBLE = 2,
 }
 
+export type CustodialChain = 'CELO' | 'TRON' | 'ONE' | 'XDC' | 'ETH' | 'MATIC' | 'KLAY' | 'BSC'
```
Comment (samuelsramko) [inline packages/shared/blockchain/evm-based/src/lib/services/evm-based.custodial.ts:43]: This should be GasPumpChain

**Review by samuelsramko — CHANGES_REQUESTED** [review]
I am missing whole functionality about gas pump here - expecially createAddress - private key is required there, it must be part of SDK. Read operations could be obtained from the OpenAPI scheme and call API directly, but not those with private keys and tx signing

File: packages/shared/blockchain/evm-based/src/lib/services/evm-based.gas.pump.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        contractAddress: evmBasedGasPump().getGasPumpFactoryContractAddress(body.chain, testnet),
+      }
+      return { params, methodName, methodSig, bodyWithContractAddress }
+    },
+
+    getGasPumpFactoryContractAddress: (chain: GasPumpChain, testnet?: boolean) => {
+      switch (chain) {
+        case Currency.CELO:
+          return testnet
+            ? '0x909ec0d70a4dA19848F6678e8a4478864dB94E0F'
```
Comment (samuelsramko) [inline packages/shared/blockchain/evm-based/src/lib/services/evm-based.gas.pump.ts:30]: use constants here and export them

**Review by samuelsramko — CHANGES_REQUESTED** [review]
We need to wait for final version of a gas pump contract to include the ABI and bytecode here

File: packages/blockchain/celo/src/lib/services/celo.tx.ts [hunk 3]
```diff
@@ -33,6 +33,12 @@ export const celoTxService = (args: { blockchain: EvmBasedBlockchain; web3: EvmB
         broadcastFunction: CeloService.celoBroadcast,
       }),
     },
+    gasPump: {
+      ...gasPump({
+        ...args,
+        broadcastFunction: BnbSmartChainService.bscBroadcast,
```
Comment (Hathoriel) [inline packages/blockchain/celo/src/lib/services/celo.tx.ts:39]: In celo you have bscBroadcast?

File: packages/blockchain/polygon/src/lib/services/polygon.tx.ts [hunk 4]
```diff
@@ -57,6 +58,12 @@ export const polygonTxService = (args: { blockchain: EvmBasedBlockchain; web3: E
         broadcastFunction: PolygonService.polygonBroadcast,
       }),
     },
+    gasPump: {
+      ...gasPump({
+        ...args,
+        broadcastFunction: BnbSmartChainService.bscBroadcast,
```
Comment (Hathoriel) [inline packages/blockchain/polygon/src/lib/services/polygon.tx.ts:64]: again in polygon bscBroadcast

File: packages/shared/blockchain/evm-based/src/lib/services/evm-based.gas.pump.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    },
+
+    getGasPumpFactoryContractAddress: (chain: GasPumpChain, testnet?: boolean) => {
+      switch (chain) {
+        case Currency.CELO:
+          return testnet ? CELO_TESTNET_CUSTODIAL_FACTORY_V2 : CELO_MAINNET_CUSTODIAL_FACTORY_V2
+        case Currency.TRON:
+          return testnet ? TRON_TESTNET_CUSTODIAL_FACTORY_V2 : TRON_MAINNET_CUSTODIAL_FACTORY_V2
+        case Currency.ONE:
+          return testnet ? ONE_TESTNET_CUSTODIAL_FACTORY_V2 : ONE_MAINNET_CUSTODIAL_FACTORY_V2
```
Comment (samuelsramko) [inline packages/shared/blockchain/evm-based/src/lib/services/evm-based.gas.pump.ts:54]: on testnet, you should throw - its not supported for now

File: packages/shared/blockchain/evm-based/src/lib/services/evm-based.gas.pump.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+export type GasPumpChain = 'CELO' | 'TRON' | 'ONE' | 'XDC' | 'ETH' | 'MATIC' | 'KLAY' | 'BSC'
+
+const range = (from: number, to: number) => to - from + 1
+
+export const CELO_TESTNET_CUSTODIAL_FACTORY_V2 = '0x5E3D93eF7C1eE50eA4536791E1441e3281f55154'
+export const CELO_MAINNET_CUSTODIAL_FACTORY_V2 = '0xd0b8460559b945aa578b81f0ab3ac998e7360332'
+
+export const ETH_TESTNET_CUSTODIAL_FACTORY_V2 = '0xA2A1BF5a83F9dAec2Dc364E1c561E937163CB613'
+export const ETH_MAINNET_CUSTODIAL_FACTORY_V2 = '0x5975906E29214339CC6D292cD6101C68f8de01e4'
```
Comment (samuelsramko) [inline packages/shared/blockchain/evm-based/src/lib/services/evm-based.gas.pump.ts:13]: this is not correct address - you have switched with testnet one @opefago

**Pull request closed without merging**